packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -0,0 +1,152 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+
+class BaseAwsSdkPlugin extends Plugin {
+  static get serviceIdentifier () {
+    const id = this.name.toLowerCase()
+    Object.defineProperty(this, 'serviceIdentifier', {
+      configurable: true,
+      writable: true,
+      enumerable: true,
+      value: id
+    })
+    return id
+  }
+
+  get serviceIdentifier () {
+    const id = this.constructor.serviceIdentifier
+    Object.defineProperty(this, 'serviceIdentifier', {
+      configurable: true,
+      writable: true,
+      enumerable: true,
+      value: id
+    })
+    return id
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    this.addSub(`apm:aws:request:start:${this.serviceIdentifier}`, ({
+      request,
+      operation,
+      awsRegion,
+      awsService,
+      serviceIdentifier
+    }) => {
+      if (!this.isEnabled(request)) {
```

Comment:
Can this be configured instead?

Reply:
Probably. I'll try to handle this with `.configure()` to make it less weird here.